### PR TITLE
Implemented structure for modifier bats

### PR DESCRIPTION
**Pull request by mythguy1226**
## Summarize what is being added
- Implemented new modifier bat that drops a shootable object for applying player modifiers
- Implemented abstract modifier class for creating any type of player modifier

## Please describe how to test
Play the scene "RobModifierBats", stun a modifier bat, and then shoot the item that is dropped! Should just spit a console log for now until we are able to design some player modifiers.

## Relevant Screenshots
image

## Issue worked on
https://bat-bots.atlassian.net/browse/BB-10?atlOrigin=eyJpIjoiNWRjYjE0OTkxMjNhNDExODgyMzhiOTZmOTUzYmMwNWIiLCJwIjoiaiJ9

## What Sprint is this due in?
Sprint 1

### CODE REVIEW

**Review by RitoxicGame — APPROVED** [review]

File: 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        if (hit.collider.gameObject == gameObject)
+        {
+            _AnimControls.PlayStunAnimation();
+            SoundManager.Instance.PlaySoundInterrupt(hitSound, 0.7f);
+
+            // Instantiate the modifier object
+            Instantiate(modifierObject, transform.position, Quaternion.identity);
+        }
+    }
+}
```
Comment (ArgentumRhodon) [inline 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs]: We might want to consider refactoring bat stun reactions so we don't need to copy the entire method to change one line. For now this is good, though.

**Review by ArgentumRhodon — APPROVED** [review]
LGTM!

File: 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs (hunk 1 above)
Comment (mythguy1226) [reply 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs]: Perhaps I could have the method instead return a boolean and then have the one line change after the fact

Comment (mythguy1226) [reply 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs]: might work this in before merging actually, it sounds like it'd be feasible

Comment (ArgentumRhodon) [reply 80s Game/Assets/Scripts/AI/StateMachine/ModifierBat/ModifierBatStateMachine.cs]: That sounds good!

**Pull request merged**